active/003-seq-feature.md

```diff
@@ -0,0 +1,108 @@
+Start date: <2015-12-18>
+Pull request: 1
+Authors: ["@RNAer", "@mortonjt"]
+Contributors: ["@RNAer", "@mortonjt", "@ebolyen", "@jairideout", "@wasade", "@gregcaporaso", "@rob-knight"]
+---
+
+# Summary
+
+Introduce an `interval_metadata` object to allow for the storage and querying of intervals.
+
+# Motivation
+
+The current implementation of `positional_metadata` in the `Sequence` class takes > 1TB memory to read in a E. coli genome with 260K features. And the `pandas` sparse data frame can alleviate the mem usage, but it is still too memory hogging. The discussion is mainly in this [thread](https://github.com/biocore/scikit-bio/issues/1159).  Storing intervals instead of `positional_metadata` would substantially save on memory.
+
+Additional benefits of having a `interval_metadata` objects include the ability to rapidly query features by interval.  IntervalTrees allow for both querying by position and ranges to retrieve all overlapping intervals.  This can be particularly helpful when dealing with layers of features (i.e. transcripts made up of exons, or operons made up of genes).
+
+# Detailed design
+
+We propose to use the `IntervalTree` object in `bx-python.`  Some benchmarks have been made for various IntervalTree data structures - in this [benchmark](https://gist.github.com/shoyer/c939325f509d7c027949), bx-python was determined to have one of the fastest query times.
+
+## `IntervalMetadata` object structure
+features : dictionary.  keys are `skbio.Feature` objects.  values are a list of tuples, where each tuple corresponds to an interval.
+intervals : bx-python `IntervalTree` object.  The keys correspond to intervals and the values correspond to a single `skbio.Feature` object.   
+
+These objects are arranged such that features can be queried by both feature attributes and intervals.
+
+## `Feature` object structure
+The feature object is more a less a immutable, hashable dictionary.  This store arbiturary information about features.  For instance, strand information, CDS, ...
+Since there is no standard in genbank and related formats for keywords, arbituary keywords can be encoded
+```python
+gene = Feature(name='sagA', type='CDS', strand='+', function='toxin')
+```
+The above code would encode for a gene, whose name is 'sagA', that is a coding region on the positive strand whose function is a toxin.
+
+## `IntervalMetadata` functions
+`constructor(features=None)`
+- `features` : dictionary.  keys are `skbio.Feature` objects.  values are a list of tuples, where each tuple corresponds to an interval.
+- Called to initialize the `IntervalMetadata` object
+
+`reverse_complement(length)`
+- `length` : int.  The length of the genome.  This is required when swapping coordinates
+- This performs a reverse complement on all the coordinates with in IntervalTree.
+
+```python
+   interval_metadata = IntervalMetadata()
+   interval_metadata.add(Feature(gene='sagB', cds=True), (3, 5))
+   iv = interval_metadata.reverse_complement(length=10)
```

Comment:
I understand what's going on: GitHub now displays outdated diff comments from email links where before the comment would have been hidden. I thought I was looking at the latest revision since I'm able to navigate to this conversation from email links. I see the updated example when viewing the pull request directly from GitHub instead of following the email link. Sorry for the confusion!